storage/innobase/handler/i_s.cc

```diff
@@ -7041,6 +7041,39 @@ i_s_dict_fill_tablespaces_encryption(
 skip:
 	DBUG_RETURN(0);
 }
+
+/** Function to fill INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION
+for temporary tablespace
+@param  thd             thread handle
+@param  space           Tablespace
+@param  table_to_fill   I_S table to fill
+@return 0 on success */
+static
+int i_s_dict_fill_temp_tablespace_encryption(THD* thd,
+                                              fil_space_t *space,
+                                              TABLE *table_to_fill)
+{
+  DBUG_ENTER("i_s_dict_fill_temp_tablespaces_encryption");
+  if (!innodb_encrypt_temporary_tables)
+    DBUG_RETURN(0);
+
+  Field **fields= table_to_fill->field;
+  OK(fields[TABLESPACES_ENCRYPTION_SPACE]->store(space->id, true));
+  OK(field_store_string(fields[TABLESPACES_ENCRYPTION_NAME],
+                        space->name));
+  OK(fields[TABLESPACES_ENCRYPTION_ENCRYPTION_SCHEME]->store(1, true));
+  OK(fields[TABLESPACES_ENCRYPTION_KEYSERVER_REQUESTS]->store(1, true));
+  OK(fields[TABLESPACES_ENCRYPTION_MIN_KEY_VERSION]->store(1, true));
+  OK(fields[TABLESPACES_ENCRYPTION_CURRENT_KEY_VERSION]->store(1, true));
+  OK(fields[TABLESPACES_ENCRYPTION_CURRENT_KEY_ID]->store(1, true));
+  OK(fields[TABLESPACES_ENCRYPTION_ROTATING_OR_FLUSHING]->store(1, true));
+
+  fields[TABLESPACES_ENCRYPTION_KEY_ROTATION_PAGE_NUMBER]->set_null();
+  fields[TABLESPACES_ENCRYPTION_KEY_ROTATION_MAX_PAGE_NUMBER]->set_null();
```

Comment:
If we are considering no rotation, shouldn't the value for `TABLESPACES_ENCRYPTION_ROTATING_OR_FLUSHING` be 0 instead of 1 ?

Looks like redo log encryption key can be rotated too and there might be some issue associated with that but let's not try to investigate that here. I will check further on it.